extra_foam/algorithms/imageproc_py.py

```diff
@@ -7,13 +7,28 @@
 Copyright (C) European X-Ray Free-Electron Laser Facility GmbH.
 All rights reserved.
 """
+import numpy as np
+
 from .imageproc import (
     nanmeanImageArray, movingAvgImageData,
     imageDataNanMask, maskImageDataNan, maskImageDataZero,
-    correctGain, correctOffset, correctDsscOffset, correctGainOffset
+    correctGain, correctOffset, correctDsscOffset, correctGainOffset,
+    binPhotons
 )
 
 
+def bin_photons(data, adu_count, out=None):
+    rank = len(data.shape)
+    if rank != 2 and rank != 3:
+        raise ValueError("Only 2D and 3D arrays are supported")
+
+    if out is None:
+        out = np.empty_like(data)
+
+    binPhotons(data, out, adu_count)
```

Comment:
Nitpick: I would suggest following the signature as observed from the other functions: `f(arr, *args, out)`. Also, using the same array when `out` is not specified.

Reply:
Changed the signature in 34f83680. I didn't default to modifying the input array in-place though, that doesn't match the behaviour of numpy/scipy so I think it'd be a little confusing to do it here.